### PR TITLE
feat: kustomize-able cluster-wide operator deploy

### PR DESCRIPTION
**Pull request by bugslifesolutions**
- remove invalid placeholder <namespace>
- separation of singleNamespace vs clusterwide RBAC
- cluster-wide manager patch
- Renamed files to better reflect responsibility

### All Submissions:

* [ x ] Have you opened an Issue before filing this PR?
* [ x ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ x ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ x ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: config/operator/clusterwide/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+bases:
+  - ../base
+
+patches:
+  - ./operator_Deployment_patch.yaml
```
Comment (rodrigovalin) [inline config/operator/clusterwide/kustomization.yaml:5]: please use `snake_case_for_filenames.yaml`

And don't forget to add a new line at the end of the file (your editor should do this for you if your configure it properly)

**Review by rodrigovalin — CHANGES_REQUESTED** [review]
Hi @bugslifesolutions 

It took me a while to understand what you are trying to do here, but finally I got it. Next time, please add a narrative/summary of your changes and goals with the patch.

From what I can see, you are creating `single-namespace` and `clusterwide` overlays on top of the Operator deployment, by moving `deploy/clusterwide` definitions into `kustomize` directory.

Additional to the narrative to why this is needed (and how you are will be doing it), it will be good if you could add 2 examples of **how to install the operator in both single-namespace and clusterwide modes**. With a good explanation of **why** and **how** it will be much easier for us to asses the value of this contribution and be able to merge it.

We always welcome contributions! Please see my comments below for the changes we need before proceeding with this patch.

**Comment by bugslifesolutions** [comment]
Thank you for your patience and coaching.  I'll make those changes and
include some kustimize examples.

**Comment by rodrigovalin** [comment]
@bugslifesolutions 

Please make sure you follow the mandatory requirements under the `All Submissions` section. The most important of them (at this stage) is that there needs to be an Issue created, discussed and agreed on a solution.

The problem you are trying to solve has not been discussed and the description of this change is very poor. It is not clear enough for us why you are making this change, what problem are you trying to solve and we have not agreed that this particular solution is the one we want for this particular problem.

At the same time, there's a failing test that you *have* to fix before we review this change.

**Comment by irajdeep** [comment]
Closing this PR, feel free to reopen if you would to continue working on this.

**Pull request closed without merging**